shared-scripts/angular-optimization/esbuild-plugin.d.ts

```diff
@@ -0,0 +1,18 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+export interface OptimizationOptions {
+  enableLinker?: {
+    ensureNoPartialDeclaration: boolean;
+    linkerOptions?: object;
+  };
+  optimize?: {
+    isSideEffectFree?: (absoluteDiskPath: string) => boolean;
+  };
+  downlevelAsyncGeneratorsIfPresent?: boolean;
+}
```

Comment:
The optimization plugin is a `.mjs` file and it needs to stay like that. The `d.ts` is just for our own documentation and potential downstream tools can even leverage it.

Long-term we might be able to replace this with an actual `.mts` file.